cEP-0016.md

```diff
@@ -0,0 +1,178 @@
+
+        if commit_parents > 1:
+            yield Result(self, 'This is a merge commit.')
+```
+
+A prototype for identifying travis commits:
+
+```python
+    def is_travis_commit(self, shortlog, body):
+        ci_regex = re.compile(r'ci skip')
+
+        if is_merge_commit and ci_regex.match(shortlog):
+            yield Result(self, 'This is travis commit.')
+```
+
+A prototype for identifying revert commits:
+
+```python
+    def is_revert_commit(self, shortlog, body):
+        revert_shortlog_regex = re.compile(r'^Revert\s"[\S+\s*]*"')
+        headline_regex = re.compile(
+            r'^This reverts commit [0-9a-f]{40}')
+        sha_regex = re.compile(r'\b[0-9a-f]{40}\b')
+
+        if not allow_revert_commits and revert_shortlog_regex.match(shortlog):
+            reverted_shortlog = shortlog.lstrip('Revert ')
+
+            if len(body) != 0:
+                body = body.strip('\n')
+                blankline = body.find('\n\n')
+                headline = body[:blankline] if blankline != -1 else body
+
+                if headline_regex.match(headline):
+                    commit_sha = re.search(sha_regex, headline)
+                    commit_sha = commit_sha.group(0)
+                    command = 'git log -n 1 ' + str(commit_sha)
+                    rstdout, rstderr = run_shell_command(command)
+
+                    if rstderr:
+                        self.err('git:', repr(stderr))
+                        yield Result(self, 'Invalid revert commit.')
+
+                    rstdout = rstdout.rstrip('\n')
+                    rpos = rstdout.find('\n')
+                    old_shortlog = rstdout[:rpos] if rpos != -1 else rstdout
+
+                    if old_shortlog == reverted_shortlog:
+                        yield Result(self, 'This is a valid revert commit.')
+
+                else:
+                    yield Result(self, 'Invalid revert commit.')
+```
+
+# Handling the special commits once identified
+
+The GitCommitBear checks the format of all the commits made to coala. With
+above changes, the special commits will be identified and treated separately
+in the following manner:
+
+Merge commits
+
+```python
+    if merge_commit:
+        yield Result(self, 'Merge commits are not allowed. Please change your commit.')
+```
+
+Travis commits
+
+```python
+    if travis_commit:
+        # skip travis build on this PR
+```
+
+Revert commits
+
+```python
+    import urllib3
+    from unidiff import PatchSet
+
+    if revert_commit:
+        http1 = remote_url.rstrp('.git') + '/pull/' + pull_index + '.diff'
+        diff1 = urllib3.PoolManager()
+
+        http2 = remote_url.rstrp('.git') + '/pull/' + pull_index + '.diff'
+        diff2 = urllib3.PoolManager()
+
+        encoding = diff.headers.getparam('charset')
+
+        patch1 = PatchSet(diff1, encoding=encoding)
+        patch2 = PatchSet(diff1, encoding=encoding)
+
+        # compare patch1 and patch2 using unidiff
+        if not(patch1) == patch2:
```

Comment:
I think practically probably we wanna do `if not patch1 == patch2:` (without the parentheses). however, this is just a draft, so it doesn't matter here ;)